components/TextFields/tests/snapshot/MDCTextFieldOutlinedControllerSnapshotTests.m

```diff
@@ -0,0 +1,213 @@
+      [[MDCTextInputControllerOutlined alloc] initWithTextInput:self.textField];
+}
+
+- (void)tearDown {
+  self.textFieldController = nil;
+  self.textField = nil;
+
+  [super tearDown];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Helpers
+
+- (void)triggerTextFieldLayout {
+  CGSize aSize = [self.textField sizeThatFits:CGSizeMake(300, INFINITY)];
+  self.textField.bounds = CGRectMake(0, 0, aSize.width, aSize.height);
+  [self.textField layoutIfNeeded];
+}
+
+- (void)generateSnapshotAndVerify {
+  [self triggerTextFieldLayout];
+
+  self.testView = [self addBackgroundViewToView:self.textField];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Tests
+
+- (void)testOutlinedTextFieldEmpty {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+#pragma marMDCTextFieldSnapshotTests - Single field tests
+
+- (void)testOutlinedTextFieldWithShortPlaceholderText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.placeholderText = MDCTextFieldSnapshotTestsPlaceholderShortTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithLongPlaceholderText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.placeholderText = MDCTextFieldSnapshotTestsPlaceholderLongTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithShortHelperText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.helperText = MDCTextFieldSnapshotTestsHelperShortTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithLongHelperText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textFieldController.helperText = MDCTextFieldSnapshotTestsHelperLongTextLatin;
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithShortErrorText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  [self.textFieldController setErrorText:MDCTextFieldSnapshotTestsErrorShortTextLatin
+                 errorAccessibilityValue:MDCTextFieldSnapshotTestsErrorShortTextLatin];
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithLongErrorText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  [self.textFieldController setErrorText:MDCTextFieldSnapshotTestsErrorLongTextLatin
+                 errorAccessibilityValue:MDCTextFieldSnapshotTestsErrorLongTextLatin];
+
+  // Then
+  [self generateSnapshotAndVerify];
+}
+
+- (void)testOutlinedTextFieldWithShortInputText {
+  // Uncomment below to recreate the golden
+  //  self.recordMode = YES;
+
+  // When
+  self.textField.text = MDCTextFieldSnapshotTestsInputShortTextLatin;
```

Comment:
The snapshot's placeholder label looks buggy on this one (there's an empty space in the outline) - do we have a bug open for this already?

Reply:
Nope, I wan't sure if that was a bug or not.

Reply:
I would suspect that using an Outlined textfield without placeholder text is a developer error. Placeholders are how we handle naming fields.

Reply:
In this case I'd expect the text field to behave well and show a complete and connected outline, so this does look like a bug.

Reply:
Filed a bug: https://github.com/material-components/material-components-ios/issues/5948